docs/reference/hooks.md

```diff
@@ -0,0 +1,344 @@
+
+Example:
+```yaml
+...
+hooks:
+  - id: trigger_my_dag
+    type: airflow
+    description: Trigger an example_dag
+    properties:
+       url: "http://localhost:8000"
+       dag_id: "example_dag"
+       username: "admin"
+       password: "{% raw %}{{{% endraw %} ENV.AIRFLOW_SECRET {% raw %}}}{% endraw %}"
+       dag_conf:
+          some: "additional_conf"
+...
+```
+
+#### Hook Record in configuration field
+
+lakeFS will add an entry to the Airflow request configuration property (`conf`) with the event that triggered the action.
+
+The key of the record will be `lakeFS_event` and the value will match the one described [here](#request-body-schema)
+
+### Lua Hooks <span>Experimental</span>{: .label }
+
+As of version 0.86.0, lakeFS supports running hooks without relying on external components using an [embedded Lua VM](https://github.com/Shopify/go-lua)
+
+Using Lua hooks, it is possible to pass a Lua script to be executed directly by the lakeFS server when an action occurs.
+
+The Lua runtime embedded in lakeFS is limited for security reasons. It provides a narrow set of APIs and functions that by default do not allow:
+
+1. Accessing any of the running lakeFS server's environment
+2. Accessing the local filesystem available the lakeFS process 
+
+#### Example of a Lua hook:
+{:.no_toc}
+
+This example will simply print out a JSON representation of the event that occurred:
+
+```yaml
+name: dump_all
+on:
+  post-commit:
+  post-merge:
+  post-create-tag:
+  post-create-branch:
+hooks:
+  - id: dump_event
+    type: lua
+    properties:
+      script: |
+        json = require("encoding/json")
+        print(json.marshal(action))
+```
+
+ A more useful example: ensure every commit contains a required metadata field:
+
+```yaml
+name: pre commit metadata field check
+on:
+pre-commit:
+    branches:
+    - main
+    - dev
+hooks:
+  - id: ensure_commit_metadata
+    type: lua
+    properties:
+      args:
+        notebook_url: {"pattern": "my-jupyter.example.com/.*"}
+        spark_version:  {}
+      script: |
+        regexp = require("regexp")
+        for k, props in pairs(args) do
+          current_value = action.commit.metadata[k]
+          if current_value == nil then
+            error("missing mandatory metadata field: " .. k)
+          end
+          if props.pattern and not regexp.match(props.pattern, current_value) then
+            error("current value for commit metadata field " .. k .. " does not match pattern: " .. props.pattern .. " - got: " .. current_value)
+          end
+        end
+```
+
+For more examples and configuration samples, check out the [examples/hooks/](https://github.com/treeverse/lakeFS/tree/master/examples/hooks) directory in the lakeFS repository.
+
+For the full set of exposed Lua APIs, usage and reasoning is available on the [proposal document](https://github.com/treeverse/lakeFS/blob/master/design/open/zero-deployment-hooks.md)
```

Comment:
Design docs usually go out of sync and are hard to track. 
The intention of users reading this section is high and they would prefer a single, clear searchable page they can view the script options, I think we should add the supported list in bullets here with short description.

Reply:
will do so in a separate PR.